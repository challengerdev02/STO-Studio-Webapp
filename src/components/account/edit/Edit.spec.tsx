import { cleanup, render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { AccountEdit } from './index';
import LiveAuctionCardImage from '../../../public/zachary-kadolph-unsplash.svg';
import MathewBall from '../../../public/mathew-ball.svg';
import BrianMcGowan from '../../../public/brian-mcgowan.svg';
import CoverImage from '../../../public/profile-cover-photo.svg';
import { Form } from 'antd';

jest.mock('next/router', () => ({
  ...jest.requireActual('next/router'), // use actual for all non-hook parts
  useParams: () => ({
    teamId: 'someTeamId',
    eventId: 'someEventId',
    broadcastId: 'someShowId',
  }),
  useRouter: () => ({
    query: {},
  }),
  useLocation: jest.fn(),
}));
describe('AccountEdit component', function () {
  afterEach(cleanup);

  const Container = () => {
    const [form] = Form.useForm();

    let props = {
      loading: true,
      onFinish: () => {},
      uploadAvatarProps: {},
      uploadCoverProps: {},
      form: form,
      $record: {},
      onValidateUsername: jest.fn(),
      isValidUsername: false,
      validatingUsername: false,

      userData: {
        id: '000001',
        username: 'Emmanuel',
        firstName: 'Emmanuel firstName',
        lastName: 'Emmanuel lastName',
        email: 'emdoe@gmail.com',
        bio: 'lore ipsum lorem',
        walletAddress: '0x658d22456633334234iF0cC',
        connections: {
          followers: '230',
          following: '84',
        },
        auctions: [
          {
            itemCoverImage: LiveAuctionCardImage,
            itemName: 'Crypto Bull Society',
            bestOffer: '0.51',
            lastOffer: '0.42',
            likes: '1.7 k',
          },
          {
            itemCoverImage: BrianMcGowan,
            itemName: 'Crypto Bull Society',
            bestOffer: '0.51',
            lastOffer: '0.42',
            likes: '1.7 k',
          },
          {
            itemCoverImage: MathewBall,
            itemName: 'Crypto Bull Society',
            bestOffer: '0.51',
            lastOffer: '0.42',
            likes: '1.7 k',
          },
          {
            itemCoverImage: BrianMcGowan,
            itemName: 'Crypto Bull Society',
            bestOffer: '0.51',
            lastOffer: '0.42',
            likes: '1.7 k',
          },
          {
            itemCoverImage: MathewBall,
            itemName: 'Crypto Bull Society',
            bestOffer: '0.51',
            lastOffer: '0.42',
            likes: '1.7 k',
          },
          {
            itemCoverImage: LiveAuctionCardImage,
            itemName: 'Crypto Bull Society',
            bestOffer: '0.51',
            lastOffer: '0.42',
            likes: '1.7 k',
          },
          {
            itemCoverImage: MathewBall,
            itemName: 'Crypto Bull Society',
            bestOffer: '0.51',
            lastOffer: '0.42',
            likes: '1.7 k',
          },
        ],
        avatar: CoverImage,
        banner: null,
        uploadStates: {
          banner: false,
          avatar: false,
        },
        loadings: {
          banner: false,
          avatar: false,
        },
      },
    };

    return <AccountEdit {...props} />;
  };

  const setup = (extraProps: Record<string, any> = {}) => {
    const utils = render(<Container {...extraProps} />);
    return {
      ...utils,
    };
  };

  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  it('should match with snapshot', () => {
    // @ts-ignore
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it('should check that the AccountEdit tab component displays', () => {
    const { getByTestId } = setup();

    expect(getByTestId('account-tab')).toBeInTheDocument();
  });

  it('should check that the Profile tab displays by default', () => {
    const { getAllByRole } = setup();
    const profileTab = getAllByRole('tab')[0];
    const notificationsTab = getAllByRole('tab')[1];
    const accountSupportTab = getAllByRole('tab')[2];

    expect(profileTab).toHaveTextContent(/profile/i);
    expect(profileTab).toHaveAttribute('aria-selected', 'true');
    expect(notificationsTab).toHaveAttribute('aria-selected', 'false');
    expect(accountSupportTab).toHaveAttribute('aria-selected', 'false');
  });

  it('should check that the Notifications tab pane is clickable, and its active when clicked', () => {
    const { getAllByRole } = setup();
    const profileTab = getAllByRole('tab')[0];
    const notificationsTab = getAllByRole('tab')[1];
    const accountSupportTab = getAllByRole('tab')[2];

    expect(notificationsTab).toHaveTextContent(/notifications/i);
    expect(notificationsTab).toHaveAttribute('aria-selected', 'false');
    expect(accountSupportTab).toHaveAttribute('aria-selected', 'false');

    fireEvent.click(notificationsTab);

    expect(notificationsTab).toHaveAttribute('aria-selected', 'true');
    expect(profileTab).toHaveAttribute('aria-selected', 'false');
    expect(accountSupportTab).toHaveAttribute('aria-selected', 'false');
  });

  it('should check that the Account Support tab pane is clickable, and its active when clicked', () => {
    const { getAllByRole } = setup();
    const profileTab = getAllByRole('tab')[0];
    const notificationsTab = getAllByRole('tab')[1];
    const accountSupportTab = getAllByRole('tab')[2];

    expect(accountSupportTab).toHaveTextContent(/account support/i);
    expect(notificationsTab).toHaveAttribute('aria-selected', 'false');
    expect(accountSupportTab).toHaveAttribute('aria-selected', 'false');

    fireEvent.click(accountSupportTab);

    expect(accountSupportTab).toHaveAttribute('aria-selected', 'true');
    expect(notificationsTab).toHaveAttribute('aria-selected', 'false');
    expect(profileTab).toHaveAttribute('aria-selected', 'false');
  });
});
