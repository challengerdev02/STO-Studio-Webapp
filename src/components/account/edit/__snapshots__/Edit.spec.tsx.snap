// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountEdit component should match with snapshot 1`] = `
<div>
  <div
    class="sc-jWaEpP kqpqkY"
  >
    <div
      class="ant-tabs ant-tabs-top ant-tabs-centered main-layout"
    >
      <div
        class="ant-tabs-nav"
        role="tablist"
      >
        <div
          class="ant-tabs-nav-wrap"
        >
          <div
            class="ant-tabs-nav-list"
            style="transform: translate(0px, 0px);"
          >
            <div
              class="ant-tabs-tab ant-tabs-tab-active"
            >
              <div
                aria-controls="rc-tabs-test-panel-profile"
                aria-selected="true"
                class="ant-tabs-tab-btn"
                id="rc-tabs-test-tab-profile"
                role="tab"
                tabindex="0"
              >
                Profile
              </div>
            </div>
            <div
              class="ant-tabs-tab"
            >
              <div
                aria-controls="rc-tabs-test-panel-notifications"
                aria-selected="false"
                class="ant-tabs-tab-btn"
                id="rc-tabs-test-tab-notifications"
                role="tab"
                tabindex="0"
              >
                Notifications
              </div>
            </div>
            <div
              class="ant-tabs-tab"
            >
              <div
                aria-controls="rc-tabs-test-panel-support"
                aria-selected="false"
                class="ant-tabs-tab-btn"
                id="rc-tabs-test-tab-support"
                role="tab"
                tabindex="0"
              >
                Account Support
              </div>
            </div>
            <div
              class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
            />
          </div>
        </div>
        <div
          class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
        >
          <button
            aria-controls="rc-tabs-test-more-popup"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-hidden="true"
            class="ant-tabs-nav-more"
            id="rc-tabs-test-more"
            style="visibility: hidden; order: 1;"
            tabindex="-1"
            type="button"
          >
            <span
              aria-label="ellipsis"
              class="anticon anticon-ellipsis"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="ellipsis"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
      <div
        class="ant-tabs-content-holder"
      >
        <div
          class="ant-tabs-content ant-tabs-content-top"
        >
          <div
            aria-hidden="false"
            aria-labelledby="rc-tabs-test-tab-profile"
            class="ant-tabs-tabpane ant-tabs-tabpane-active"
            id="rc-tabs-test-panel-profile"
            role="tabpanel"
            tabindex="0"
          >
            <div
              class="content-profile"
            >
              <h1
                class="ant-typography title"
              >
                <span>
                  Profile Settings
                </span>
                <button
                  class="ant-btn ant-btn-round"
                  type="button"
                >
                  <span
                    aria-label="eye"
                    class="anticon anticon-eye"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="eye"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
                      />
                    </svg>
                  </span>
                  <span>
                    Preview
                  </span>
                </button>
              </h1>
              <form
                autocomplete="off"
                class="ant-form ant-form-vertical"
                id="dynamic_form_nest_item"
              >
                <div
                  class="ant-row"
                >
                  <div
                    class="ant-col left-fields ant-col-md-24 ant-col-lg-12 ant-col-xl-12 ant-col-xxl-12"
                  >
                    <div
                      class="ant-row ant-form-item"
                    >
                      <div
                        class="ant-col ant-form-item-label"
                      >
                        <label
                          class="ant-form-item-required"
                          for="dynamic_form_nest_item_username"
                          title="Username"
                        >
                          Username
                        </label>
                      </div>
                      <div
                        class="ant-col ant-form-item-control"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <input
                              class="ant-input"
                              id="dynamic_form_nest_item_username"
                              placeholder="Emmanuel"
                              type="text"
                              value=""
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="ant-row ant-form-item"
                    >
                      <div
                        class="ant-col ant-form-item-label"
                      >
                        <label
                          class=""
                          for="dynamic_form_nest_item_firstName"
                          title="First Name"
                        >
                          First Name
                        </label>
                      </div>
                      <div
                        class="ant-col ant-form-item-control"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <input
                              class="ant-input"
                              id="dynamic_form_nest_item_firstName"
                              placeholder="Emmanuel firstName"
                              type="text"
                              value=""
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="ant-row ant-form-item"
                    >
                      <div
                        class="ant-col ant-form-item-label"
                      >
                        <label
                          class=""
                          for="dynamic_form_nest_item_lastName"
                          title="Last Name"
                        >
                          Last Name
                        </label>
                      </div>
                      <div
                        class="ant-col ant-form-item-control"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <input
                              class="ant-input"
                              id="dynamic_form_nest_item_lastName"
                              placeholder="Emmanuel lastName"
                              type="text"
                              value=""
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="ant-row ant-form-item"
                    >
                      <div
                        class="ant-col ant-form-item-label"
                      >
                        <label
                          class=""
                          title="Wallet Address"
                        >
                          Wallet Address
                        </label>
                      </div>
                      <div
                        class="ant-col ant-form-item-control"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <span
                              class="ant-input-group ant-input-group-compact"
                            >
                              <input
                                class="ant-input ant-input-disabled"
                                disabled=""
                                placeholder="0x658d22456633334234iF0cC"
                                style="width: calc(100% - 200px);"
                                type="text"
                                value="0x658d22456633334234iF0cC"
                              />
                              <button
                                class="ant-btn ant-btn-icon-only"
                                type="button"
                              >
                                <span
                                  aria-label="copy"
                                  class="anticon anticon-copy"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="copy"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M832 64H296c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h496v688c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V96c0-17.7-14.3-32-32-32zM704 192H192c-17.7 0-32 14.3-32 32v530.7c0 8.5 3.4 16.6 9.4 22.6l173.3 173.3c2.2 2.2 4.7 4 7.4 5.5v1.9h4.2c3.5 1.3 7.2 2 11 2H704c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32zM350 856.2L263.9 770H350v86.2zM664 888H414V746c0-22.1-17.9-40-40-40H232V264h432v624z"
                                    />
                                  </svg>
                                </span>
                              </button>
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-col right-fields ant-col-md-24 ant-col-lg-12 ant-col-xl-12 ant-col-xxl-12"
                  >
                    <div
                      class="ant-row ant-form-item"
                    >
                      <div
                        class="ant-col ant-form-item-label"
                      >
                        <label
                          class=""
                          title="Profile Image"
                        >
                          Profile Image
                        </label>
                      </div>
                      <div
                        class="ant-col ant-form-item-control"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <span
                              class=""
                            >
                              <div
                                class="ant-upload ant-upload-select ant-upload-select-text"
                              >
                                <span
                                  class="ant-upload"
                                  role="button"
                                  tabindex="0"
                                >
                                  <input
                                    accept=""
                                    style="display: none;"
                                    type="file"
                                  />
                                  <div>
                                    <div
                                      class="sc-bxDdli cNpvNS"
                                      height="80px"
                                      width="80px"
                                    >
                                      <i
                                        class="mc-pencil-fill"
                                        style="font-size: 1.5em;"
                                      />
                                    </div>
                                    <span
                                      class="ant-avatar ant-avatar-circle ant-avatar-image"
                                      data-testid="user-avatar"
                                      style="width: 80px; height: 80px; line-height: 80px; font-size: 18px; transition: 2s; opacity: 1;"
                                    >
                                      <img
                                        src="/test-file-stub"
                                      />
                                    </span>
                                  </div>
                                </span>
                              </div>
                              <div
                                class="ant-upload-list ant-upload-list-text"
                              />
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="ant-row ant-form-item"
                    >
                      <div
                        class="ant-col ant-form-item-label"
                      >
                        <label
                          class=""
                          title="Profile Banner"
                        >
                          Profile Banner
                        </label>
                      </div>
                      <div
                        class="ant-col ant-form-item-control"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <span
                              class=""
                            >
                              <div
                                class="ant-upload ant-upload-select ant-upload-select-text"
                              >
                                <span
                                  class="ant-upload"
                                  role="button"
                                  tabindex="0"
                                >
                                  <input
                                    accept=""
                                    style="display: none;"
                                    type="file"
                                  />
                                  <div
                                    class="sc-bxDdli bAKGoq"
                                  >
                                    <i
                                      class="mc-pencil-fill mc-2x"
                                    />
                                  </div>
                                  <div
                                    class="ant-image"
                                    style="width: 150px;"
                                  >
                                    <img
                                      alt="Cover image"
                                      class="ant-image-img cover-image"
                                      src="/test-file-stub"
                                      style="transition: 2s; opacity: 1;"
                                    />
                                  </div>
                                </span>
                              </div>
                              <div
                                class="ant-upload-list ant-upload-list-text"
                              />
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row"
                >
                  <div
                    class="ant-row ant-form-item"
                  >
                    <div
                      class="ant-col ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <button
                            class="ant-btn ant-btn-primary ant-btn-round"
                            disabled=""
                            style="display: flex; align-items: center; justify-content: center; gap: 4px;"
                            type="submit"
                          >
                            <span>
                              Save
                            </span>
                            <span
                              class="sc-cLpAjG dBHYcG"
                              role="1"
                            >
                              <svg
                                fill="none"
                                height="14"
                                viewBox="0 0 14 14"
                                width="14"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="M7 0.333374C6.63181 0.333374 6.33333 0.631851 6.33333 1.00004V3.00004C6.33333 3.36823 6.63181 3.66671 7 3.66671C7.36819 3.66671 7.66667 3.36823 7.66667 3.00004V1.00004C7.66667 0.631851 7.36819 0.333374 7 0.333374Z"
                                  fill="#FCFCFD"
                                />
                                <path
                                  d="M7 10.3334C6.63181 10.3334 6.33333 10.6319 6.33333 11V13C6.33333 13.3682 6.63181 13.6667 7 13.6667C7.36819 13.6667 7.66667 13.3682 7.66667 13V11C7.66667 10.6319 7.36819 10.3334 7 10.3334Z"
                                  fill="#FCFCFD"
                                />
                                <path
                                  d="M13 6.33337C13.3682 6.33337 13.6667 6.63185 13.6667 7.00004C13.6667 7.36823 13.3682 7.66671 13 7.66671H11C10.6318 7.66671 10.3333 7.36823 10.3333 7.00004C10.3333 6.63185 10.6318 6.33337 11 6.33337H13Z"
                                  fill="#FCFCFD"
                                />
                                <path
                                  d="M3.66667 7.00004C3.66667 6.63185 3.36819 6.33337 3 6.33337H0.999999C0.631809 6.33337 0.333332 6.63185 0.333332 7.00004C0.333332 7.36823 0.631809 7.66671 0.999999 7.66671H3C3.36819 7.66671 3.66667 7.36823 3.66667 7.00004Z"
                                  fill="#FCFCFD"
                                />
                                <path
                                  d="M10.7713 2.28606C11.0316 2.02571 11.4537 2.02571 11.7141 2.28606C11.9744 2.54641 11.9744 2.96852 11.7141 3.22887L10.2999 4.64308C10.0395 4.90343 9.61741 4.90343 9.35706 4.64308C9.09671 4.38273 9.09671 3.96062 9.35706 3.70027L10.7713 2.28606Z"
                                  fill="#FCFCFD"
                                />
                                <path
                                  d="M4.64295 9.35703C4.3826 9.09668 3.96049 9.09668 3.70014 9.35703L2.28593 10.7712C2.02558 11.0316 2.02558 11.4537 2.28593 11.714C2.54628 11.9744 2.96839 11.9744 3.22874 11.714L4.64295 10.2998C4.9033 10.0395 4.9033 9.61737 4.64295 9.35703Z"
                                  fill="#FCFCFD"
                                />
                                <path
                                  d="M11.7141 10.7713C11.9744 11.0317 11.9744 11.4538 11.7141 11.7141C11.4537 11.9745 11.0316 11.9745 10.7713 11.7141L9.35704 10.2999C9.09669 10.0396 9.09669 9.61745 9.35704 9.3571C9.61739 9.09675 10.0395 9.09675 10.2998 9.3571L11.7141 10.7713Z"
                                  fill="#FCFCFD"
                                />
                                <path
                                  d="M4.64301 4.64299C4.90336 4.38265 4.90336 3.96054 4.64301 3.70019L3.2288 2.28597C2.96845 2.02562 2.54634 2.02562 2.28599 2.28597C2.02564 2.54632 2.02564 2.96843 2.28599 3.22878L3.70021 4.64299C3.96056 4.90334 4.38267 4.90334 4.64301 4.64299Z"
                                  fill="#FCFCFD"
                                />
                              </svg>
                            </span>
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </form>
            </div>
          </div>
          <div
            aria-hidden="true"
            aria-labelledby="rc-tabs-test-tab-notifications"
            class="ant-tabs-tabpane"
            id="rc-tabs-test-panel-notifications"
            role="tabpanel"
            style="display: none;"
            tabindex="-1"
          />
          <div
            aria-hidden="true"
            aria-labelledby="rc-tabs-test-tab-support"
            class="ant-tabs-tabpane"
            id="rc-tabs-test-panel-support"
            role="tabpanel"
            style="display: none;"
            tabindex="-1"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
